tests/unit/vectors.frag

```diff
@@ -23,7 +23,16 @@ float withExtraComponents() {
     vec2 v2 = vec2(1);
     vec3 v3 = vec3(1, 2, v2.x);
     vec4 v4 = vec4(v2, v3.xy);
-    return v2.x + v3.x + v4.x;
+    vec3 v5 = vec3(1, v3.rg);
```

Comment:
> So I think that a.xy can become a whenever it's the last argument and there's more than one argument.

Again, assuming I understand you correctly -- that is also properly handled by this PR, and tested via these cases here?

Reply:
alright, I think we can merge

Thank!